scripts/Metacalibration/metacal.py

```diff
@@ -0,0 +1,189 @@
+	minus_g1 = minus_g1_gal.drawImage(scale=pixel_scale, method='no_pixel')
+
+	plus_g2 = plus_g2_gal.drawImage(scale=pixel_scale, method='no_pixel')
+	minus_g2 = minus_g2_gal.drawImage(scale=pixel_scale, method='no_pixel')
+
+	# plus_moments_g1 = plus_g1.FindAdaptiveMom()
+	# minus_moments_g1 = minus_g1.FindAdaptiveMom()
+	#
+	# plus_moments_g2 = plus_g2.FindAdaptiveMom()
+	# minus_moments_g2 = minus_g2.FindAdaptiveMom()
+
+	psf_shearestimator_image = psf_shearestimator.drawImage(scale=pixel_scale)
+
+	plus_moments_g1 = galsim.hsm.EstimateShear(plus_g1, psf_shearestimator_image, shear_est=shearestimator)
+	minus_moments_g1 = galsim.hsm.EstimateShear(minus_g1, psf_shearestimator_image, shear_est=shearestimator)
+	plus_moments_g2 = galsim.hsm.EstimateShear(plus_g2, psf_shearestimator_image, shear_est=shearestimator)
+	minus_moments_g2 = galsim.hsm.EstimateShear(minus_g2, psf_shearestimator_image, shear_est=shearestimator)
+
+	# plus_shape_g1 = plus_moments_g1.observed_shape
+	# minus_shape_g1 = minus_moments_g1.observed_shape
+	#
+	# plus_shape_g2 = plus_moments_g2.observed_shape
+	# minus_shape_g2 = minus_moments_g2.observed_shape
+
+	e1_plus_g1 = plus_moments_g1.corrected_e1
+	e2_plus_g1 = plus_moments_g1.corrected_e2
+
+	e1_minus_g1 = minus_moments_g1.corrected_e1
+	e2_minus_g1 = minus_moments_g1.corrected_e2
+
+	e1_plus_g2 = plus_moments_g2.corrected_e1
+	e2_plus_g2 = plus_moments_g2.corrected_e2
+
+	e1_minus_g2 = minus_moments_g2.corrected_e1
+	e2_minus_g2 = minus_moments_g2.corrected_e2
+
+	# calculating the shear response matrix R
+	R_11 = (e1_plus_g1 - e1_minus_g1) / (2 * cshear_delta_g1)
+	R_12 = (e2_plus_g1 - e2_minus_g1) / (2 * cshear_delta_g1)
+	R_21 = (e1_plus_g2 - e1_minus_g2) / (2 * cshear_delta_g2)
+	R_22 = (e2_plus_g2 - e2_minus_g2) / (2 * cshear_delta_g2)
+
+	R = np.array([[R_11, R_12],[R_21, R_22]])
+
+	# Calculating shape of reconvolved_no_shear to test accuracy of shear response
+	noshear_image = reconvolved_noshear.drawImage(scale=pixel_scale, method='no_pixel')
+	noshear_moments = galsim.hsm.EstimateShear(noshear_image, psf_shearestimator_image, shear_est=shearestimator)
+	noshear_e1 = noshear_moments.corrected_e1
+	noshear_e2 = noshear_moments.corrected_e2
+
+
+
+	return R, noshear_e1, noshear_e2
+
+
+def metacalibration(original_gal, oshear_delta_g1, oshear_delta_g2, true_psf, psf_deconvolve, psf_reconvolve, psf_shearestimator, cshear_delta_g1, cshear_delta_g2, shearestimator, pixel_scale):
+	"""
+	Takes in an observed galaxy profile, the deconvolution and reconvolution PSFs,
+	and the amounts by which to vary g1 and g2, then performs metacalibration based
+	on these parameters. The function prints and returns the shear response matrix R
+	as a numpy array.
+
+	tuples are added to the results list
+	"""
+	cosmic_sheared_galaxy = original_gal.shear(g1=oshear_delta_g1, g2=oshear_delta_g2)
+	observed_galaxy_profile = generate_observed_galaxy(cosmic_sheared_galaxy, true_psf, oshear_delta_g1, oshear_delta_g2)
```

Comment:
You apply twice oshear_delta_gi here: a first time in `original_gal.shear` and a second time in `generate_observed_galaxy` if you look the definition of the function. Is it what you want to do?

I guess you want to do only that: 

```suggestion
	observed_galaxy_profile = generate_observed_galaxy(original_gal, true_psf, oshear_delta_g1, oshear_delta_g2)
```

Reply:
@alexsunthesun and @sidneymau  I guess this is here that might be a part of the issue. Am I wrong?

Reply:
I think you're right! That's definitely not what I was intending to do. I'll fix that and then regenerate some data to see if that solves the problem. Thanks!